lang/java/java.talon

```diff
@@ -0,0 +1,56 @@
+mode: user.java
+mode: user.auto_lang
+and code.language: java
+-
+tag(): user.code_operators
+tag(): user.code_comment
+tag(): user.code_block_comment
+tag(): user.code_generic
+
+action(user.code_state_import):
+    insert("import ")
+
+# Types Commands
+type {user.code_types}: 
+    insert(code_types)
+    key("space")
+
+type ({user.java_common_classes}|{user.java_generic_data_structures}): 
```

Comment:
I'd say maybe swap your .py stuff to put all the unboxed stuff into code_types along with the common classes and generic data structures, and then you could just fall back to the 'type *' command defined in programming.talon.